lib/Interpreter/Interpreter.cpp

```diff
@@ -776,6 +803,9 @@ namespace cling {
   Interpreter::process(const std::string& input, Value* V /* = 0 */,
                        Transaction** T /* = 0 */,
                        bool disableValuePrinting /* = false*/) {
+    if (!isInSyntaxOnlyMode() && m_Opts.CompilerOpts.CUDAHost)
```

Comment:
Would it be better to make these `virtual`, and have a derived `Interpreter`for the CUDAHost case?

Reply:
Yes, it would be useful to use `virtual` to improve code quality. I think a derived class would make it easier to see what changes are necessary for the CUDA mode.

I used the "if" solution because I wasn't sure if the `virtual` statement could negatively affect performance, but I see that the destructor is already `virtual`. So this is not a problem and I see no further disadvantage. 

I will check a possible solution in a future PR.

Reply:
Good arguments - thanks, I'm counting on you here :-)